pt-liferay-solutions/liferay-portal
# LPS-140169 [CP 2.0] Key details - DXP

## How it was developed?

It was created a modal that opens when the user clicks on the activation key to show more details of this key (check this link https://github.com/brianchandotcom/liferay-portal/pull/113037). 

The modal has three columns (Environment Group, Server Group and Status) that was made with a grid system to be aligned. 

![Screenshot from 2022-02-04 11-53-34](https://user-images.githubusercontent.com/80365844/152598072-0a93c281-6fa1-4021-9822-a053df760104.png)

## **What has been affected by the development ?**

- We had to add a new property on the Activation Keys Table rows to pass the activation key data to modal and to set a new state to show the modal when we click on the key.
- We created a new state to pass the Activation Key data to modal (currentActivationKey).

## **Adding data mock up to test**

You can follow the same steps that we use to see the Activation Keys Table here https://github.com/brianchandotcom/liferay-portal/pull/113037

## **Test Plan**

1. Click on a key to open the modal:
    
![Screenshot from 2022-02-04 11-53-34](https://user-images.githubusercontent.com/80365844/152598325-4e845e68-06cf-47de-8588-339043d629e5.png) 
 
2. Try different types of status and data on the table to see how it changes inside the modal too;
    
![Screenshot from 2022-02-04 13-39-10](https://user-images.githubusercontent.com/80365844/152598437-eb52cf69-9a6d-42bb-af75-46d4d28ff359.png)

![Screenshot from 2022-02-04 13-39-16](https://user-images.githubusercontent.com/80365844/152598450-226bc5e2-aacc-49fb-9c8f-e58c2ccffb49.png)
    
3. Try On-Premise and Virtual Cluster to see how the columns changes (Virtual Cluster doesn’t have Host Name, IP and Mac Addresses, should only show the number of cluster nodes);
    
![Screenshot from 2022-02-04 13-39-31](https://user-images.githubusercontent.com/80365844/152598503-ce1392fe-d5b8-4b39-8330-048d091c168a.png)
    
4. Choose an On-Premise key type that have an Host Name or IP and Mac Addresses to see the copy icon on the modal;
5. Click on one of these icons to see the success message confirming the type of data you copied;
    
![Screenshot from 2022-02-04 13-40-00](https://user-images.githubusercontent.com/80365844/152598558-f2988a35-b209-4db4-a0a9-01e453179073.png)
    
6. The alert message has autoClose or can be closed with the ‘X’ icon;
7. Try to paste the data (CTRL + V) to see that you had copied with success;
8. You can close the modal both with the ‘Close’ button or with the ‘X’ button on the top of the modal. 

P.s: The download button is not functioning yet because it makes part of the LPS-146803 subtask that is in development.
